### PR TITLE
Update Gopkg.lock and cleanup Makefile

### PR DESCRIPTION
**Pull request by detiber**
**What this PR does / why we need it**:
- Cleans up Gopkg.lock diff after running make targets
- Fixes target name for `create-cluster-ha`
- Leverages bazel for building `binaries-dev` target instead of rebuilding with `go get`

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -211,8 +211,7 @@ release-artifacts: ## Build release artifacts
 ## --------------------------------------
 
 .PHONY: binaries-dev
-binaries-dev: ## Builds and installs all development binaries using go get
-	go get -v ./...
+binaries-dev: generate manager clusterawsadm clusterctl
```
Comment (chuckha) [inline Makefile]: this installs the binaries to GOPATH?

Comment (detiber) [reply Makefile]: Yes: https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/970144de689f92f1b7d42c34da87b218f1f01dc4/Makefile#L178-L191

**Pull request merged**